### PR TITLE
LUC-1641: Adding duplication handler 

### PR DESCRIPTION
**Pull request by karl00schmidt**
At the moment, the duplication of an object results in an identical copy. This is the intended consequence, however, it can become hard to distinguish between a copied element, and the original, especially in id based systems. This PR should address this.

### CODE REVIEW

**Review by beingforthebenefit — COMMENTED** [review]
Makes sense to me. Will approve after we get your local instance up and test this thing.

File: packages/@tinacms/fields/src/plugins/GroupListFieldPlugin.tsx [hunk 1]
```diff
@@ -157,6 +156,7 @@ const Item = ({ tinaForm, field, index, item, label, ...p }: ItemProps) => {
     const newItem = {
       ...deepCopy,
       name: item.name ? `${item.name} (copy)` : undefined,
+      isCopy: true,
```
Comment (jcuna) [inline packages/@tinacms/fields/src/plugins/GroupListFieldPlugin.tsx]: Isn't this a tina's `Field` type or `GroupFieldDefinititon` type? Shouldn't we update the interface as for that as well?

**Review by jcuna — CHANGES_REQUESTED** [review]
Checkout my comment please

**Comment by karl00schmidt** [comment]
Updated to use handler instead

File: packages/@tinacms/fields/src/plugins/GroupListFieldPlugin.tsx (hunk 1 above)
Comment (jcuna) [reply packages/@tinacms/fields/src/plugins/GroupListFieldPlugin.tsx]: We don't need this anymore

**Review by jcuna — CHANGES_REQUESTED** [review]
Looks good other than still having `isCopy`, which is not needed anymore

File: packages/@tinacms/fields/src/plugins/GroupListFieldPlugin.tsx [hunk 2]
```diff
@@ -62,7 +63,7 @@ interface GroupFieldDefinititon extends Field {
      */
     label?: string
   }
-  allowItemDuplication?: boolean
+  onItemDuplication?: (name: string, index: number, value: any, callback: Mutators['insert']) => void
```
Comment (jcuna) [inline packages/@tinacms/fields/src/plugins/GroupListFieldPlugin.tsx]: Great use of types

**Review by jcuna — APPROVED** [review]
Awesome stuff man. Thanks a lot!

File: packages/@tinacms/fields/src/plugins/GroupListFieldPlugin.tsx (hunk 2 above)
Comment (jcuna) [reply packages/@tinacms/fields/src/plugins/GroupListFieldPlugin.tsx]: Actually, can we rename that from callback to `formInsert` or something more relevant? callback is too generic

**Review by jcuna — APPROVED** [review]

**Pull request merged**